File: ClientApp/app/models/reservation.ts
import {Vehicle } from './vehicle';
import {ParkingLot } from './parkingLot';

export interface Reservation {
    id: number;
    takeDate: Date;
    returnDate: Date; 
    name: string; 
    surname: string;
    phoneNumber: string;
    email: string;
    vehicle: Vehicle;
    parkingLot: ParkingLot;
    finished: boolean;
    fuel: string;
    washed: boolean;
    comment: string;
    
  }
  
  export interface SaveReservation {
    id: number;
    takeDate: Date;
    returnDate: Date; 
    name: string; 
    surname: string;
    phoneNumber: string;
    email: string;
    vehicleId: number;
    parkingLotId: number;
    finished: boolean;
    fuel: string;
    washed: boolean;
    comment: string;
  }

File: Core/Models/Reservation.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace vega.Core.Models

{
     [Table("Reservations")]
    public class Reservation
    {
        public int Id { get; set; }
        [Required]
        public DateTime TakeDate { get; set; }
        [Required]
        public DateTime ReturnDate { get; set; }
        [Required]
        [StringLength(255)]
        public string Name{get;set;}
        [Required]
        [StringLength(255)]
        public string Surname{get;set;}
        [Required]
        [StringLength(255)]
        public string PhoneNumber{get;set;}
        [Required]
        [EmailAddress]
        public string Email{get;set;}
        
        public int VehicleId { get; set; }
        public Vehicle Vehicle { get; set; }
        public int ParkingLotId { get; set; }
        public ParkingLot ParkingLot { get; set; }
        
        public bool Finished{get;set;}
        public string Fuel{get;set;}
        public bool Washed{get;set;}
        public string Comment{get;set;}

    }
}